facet-reflect/src/poke/enum_.rs

```diff
@@ -146,6 +149,19 @@ impl<'mem> PokeEnumNoVariant<'mem> {
             selected_variant: variant_index,
         })
     }
+
+    /// Try to find the variant by name.
+    pub fn variant_by_name(&self, name: &str) -> Option<&Variant> {
+        self.def
+            .variants
+            .iter()
+            .find(|variant| variant.name == name)
+    }
```

Comment:
I think ideally the `facet-reflect` API will evolve to have these methods return "Slots" that can be set.

That's what I was doing initially, but it took me a while to figure out what the right shape of that would be. No pun intended.